### DISCUSSION TITLE
Formatting dipcall vcf to use in truvari

### DISCUSSION THREAD

**Discussion by mchaisso** [spiralgenetics/truvari#161] [General]
Hi Adam,
  I need to format a dipcall vcf for use in truvari. The output of dipcall is follows:
chr1    181157  .       ggcgcagagaggcgcgccgcgccggcgcaggcgcagagaggcgcgccgcgccggcgcaggcgcagagaggcgcgcctcgccggcgcaggcgcagagaggcgcgccgggccggcgcaggcgcagagaggc       G       30      .       .       GT:AD   1|1:0,2

This has problems: missing SVLEN, and has multiallelic calls. I added SVLEN based on the first alt allele length, and made the multiallelic genotypes reference only 0 or 1 genotypes, but there may be a problem still. When I perform benchmarking, nearly identical calls (insertions) between the bench and compare are showing up as fp's. 

Thanks in advance,
-Mark

**Comment by ACEnglish** [comment]
Hello,

Truvari doesn't handle multi-allelic VCFs ([wiki](https://github.com/ACEnglish/truvari/wiki/Multi-allelic-VCFs)). If you're using version 4.0+ they should raise an error and exit early. To fix this, simply run `bcftools norm -m-any`. Given that you're using dipcall, and presumably have haplotype information, you may also want to prevent `norm` from maybe destroying that haplotype information with `-N/--do-not-normalize`.
Also, since you'll be splitting the records, you may be interested in using `--pick ac` ([details](https://github.com/ACEnglish/truvari/wiki/bench#controlling-the-number-of-matches))

For the INFO fields you'd like to add, simply run `truvari anno svinfo` which adds SVTYPE and SVLEN INFO fields ([details](https://github.com/ACEnglish/truvari/wiki/anno#truvari-anno-svinfo)).

Please let me know if this fixes your problems or you have more questions.

Have a great day,
~/Adam English
